index.js

```diff
@@ -435,11 +435,18 @@
          * Utility function to iterate over the list and call the fn provided
          * on each node, or element, of the list
          *
-         * param {object} fn The function to call on each node of the list
+         * @param {object} fn The function to call on each node of the list
+         * @param {bool} reverse Use or not reverse iteration (tail to head), default to false
          */
-        forEach: function(fn) {
+        forEach: function(fn, reverse) {
+          reverse = reverse || false;
```

Comment:
This convention is something that will be fixed with es2015 default params.  I am not a big fan of changing the value of a local param; however, this is fairly common and I do see it a lot.  Personally, I just create another local var (and initialize to the local param or default value) to use within the function.  It's ok in this case...